File: src/dataset/javascript/prime-factors_exercism-ThomasZumsteg/claude/ClaudeSonnet4_prime-factors_v1.js
function primeFactors(remainder) {
	/* Finds all prime factors of a number */
	// Finds factors smallest to largest adding to the list of factors
	var factors = [];
	
	// Handle factor 2 separately
	while(remainder % 2 === 0) {
		factors.push(2);
		remainder >>>= 1; // Bit shift division by 2
	}
	
	// Check odd factors starting from 3
	var factor = 3;
	var limit = Math.sqrt(remainder);
	while(factor <= limit) {
		while(remainder % factor === 0) {
			factors.push(factor);
			remainder /= factor;
			limit = Math.sqrt(remainder); // Update limit as remainder shrinks
		}
		factor += 2; // Skip even numbers
	}
	
	if(remainder > 1)
		factors.push(remainder);
	return factors;
}

module.exports = {for: primeFactors};

File: src/dataset/python/prime-factors_exercism-ThomasZumsteg/claude/ClaudeSonnet4_prime_factors_v1.py
"""Finds the prime factors of a number"""

def prime_factors(number):
    """Finds prime factors of a number"""
    if number <= 1:
        return []
    
    factors = []
    
    # Handle factor 2 separately to avoid checking even numbers later
    while number % 2 == 0:
        factors.append(2)
        number //= 2
    
    # Check odd factors only, starting from 3
    factor = 3
    while factor * factor <= number:
        while number % factor == 0:
            factors.append(factor)
            number //= factor
        factor += 2
    
    # If number is still greater than 1, it's a prime factor
    if number > 1:
        factors.append(int(number))
    
    return factors